src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -872,3 +873,16 @@ func statefulSetKind() string {
 func (r *StatefulSetResource) fullConfiguratorImage() string {
 	return fmt.Sprintf("%s:%s", r.configuratorSettings.ConfiguratorBaseImage, r.configuratorSettings.ConfiguratorTag)
 }
+
+// Version returns the cluster version specified in the spec image.
+func (r *StatefulSetResource) Version() string {
```

Comment:
I think this misses a bit the point. The status should not just mirror the spec at any point in time. One if its main purpose is to signal the completion of an upgrade.

So, the first thing we do when users change the `spec.version` of the redpanda cluster is to change  `spec.template.spec.containers[0].image` of the sts, so I think (I may be wrong) the `status.version` of the cluster will reflect that value at the next reconcile cycle.

While the idea was: when the user edits `spec.version`, we keep `status.version` as it is, until the upgrade is fully completed.. e.g. all pods belonging to the statefulset have been changed to use the new image, which does not happen immediately but after a looong restart process (for some production clusters). We may also check their readiness status, but we may want to add another indicator for that in another PR (e.g. a condition).

This way we have a clear indication that we're doing an upgrade from version `status.version` to `spec.version` during the process and a way to identify cases when the upgrade remains stuck.